packages/hashi/README.md

```diff
@@ -22,3 +22,104 @@ Step 1: Install Hashi package deps (run from Kolibri root)
+
+### What would the user experience?
+
+When a user has permissions to access a custom channel, and they click on it in the main learn tab, rather than viewing "normal Kolibri," they will experience a full-screen HTML5 app. Through the functionality provided by the API, they can interact with content within the app, and hashi will act as a "bridge" - collecting information about the user's interactions (such as clicking on a link), requesting data from the Kolibri database based on the type of interaction, and passing the requested data back to the app. One "out-of-the-box" user interaction is the `navigateTo()` function, which opens  a modal that displays a content node. For other data fetching requests, the app, not Kolibri, has the responsibilty of determining what to do with that data.
+
+Below: A gif that demonstrates navigating into a normal Kolibri channel and then the difference of navigation into a custom channel. Once in the custom channel, clicking a link to a content node opens a modal that then displays the content.
+
+![updated-fullscreen-render](https://user-images.githubusercontent.com/17235236/118153282-b88b0a00-b3e3-11eb-87ed-8657200ccedb.gif)
+
+
+### Basic API
+
+Detailed specifications about the API are in `packages/hashi/kolibri.js` written with JSDoc, but a brief overview of the functions that are available and how to use them is detailed below.
+
+- Access the API using `window.kolibri.[function]`
+
+Functions:
+- `getContentByFilter(options)`
+    Options is an object that determines what to filter by
+    Returns matching content node objects
+- `getContentById(id)`
+    Returns a single matching content node object
+- `searchContent(options)`
+    Options is an object that determines what to search by, using keywords and topics
+    Returns matching content node objects
+- `themeRenderer(options)`
+    Options is an object that determines the "theme" (colors) used in the modals
+- `navigateTo(nodeId, context)`
+    Either launches the render overlay for a content node within a custom context or loads a topic or channel in main Kolibri
+- `updateContext(context)`
+    Updates the context object that is encoded in the URL
+- `getContext()`
+    Gets the current context object that is encoded in the URL
+- `getVersion()`
+    Gets the version of Kolibri (and thus API) that is available
+
+
+### Data Flow
+
+The HTML5 App, using window.kolibri, calls a Hashi Kolibri API function. The general API pattern is set up through promises, which allows us to manage asynchronous data requests.
+
+`sendMessageAwaitReply()`, which is defined in mediator.js, is the core of how these requests are managed. The function registers and removes message handlers, which in turn use the postMessage API to communicate either locally (to `window`) or to `remote`.
```

Comment:
I think it is useful to document this, but I feel like documenting the APIs for someone developing against Hashi should be separated from documenting these implementation details for someone developing hashi.